joblib/numpy_pickle_utils.py

```diff
@@ -299,7 +299,7 @@ def __init__(self, filename, mode="rb", compresslevel=9):
         else:
             raise ValueError("Invalid mode: %r" % (mode,))
 
-        if isinstance(filename, (str, bytes)):
+        if isinstance(filename, (str, bytes, (sys.version_info < (3,) and unicode) or str)):
```

Comment:
Using `isinstance(filename, compat._bytes_or_unicode)` is the right way to go about it I reckon.

It would be good to add a regression test in joblib/test/test_numpy_pickle_utils.py to make sure that b'myfilename', 'myfilename' and u'myfilename' can be used in BinaryZlibFile.